### PR TITLE
GODRIVER-2434 Use the unified spec test runner for APM tests.

### PR DESCRIPTION
**Pull request by benjirewis**
GODRIVER-2434

## Summary

Use the unified test runner for APM tests. Note that we cannot sync the "logging" tests yet, as we have not implemented logging in the driver.

## Background & Motivation

This will be helpful for parallelizing the testing suite, as there were namespace conflicts between the legacy and unified command monitoring tests.

### CODE REVIEW

**Review by prestonvasquez — APPROVED** [review]
LGTM!

File: testdata/command-monitoring/pre-42-server-connection-id.json [hunk 1]
```diff
@@ -98,4 +98,4 @@
       ]
     }
   ]
-}
```
Comment (qingyang-hu) [inline testdata/command-monitoring/pre-42-server-connection-id.json:101]: Was the new line removed accidentally or intentionally?

**Review by qingyang-hu — APPROVED** [review]
Looks good except some small questions.

File: testdata/command-monitoring/pre-42-server-connection-id.json (hunk 1 above)
Comment (benjirewis) [reply testdata/command-monitoring/pre-42-server-connection-id.json:101]: I copied the contents of the [`monitoring`](https://github.com/mongodb/specifications/tree/master/source/command-logging-and-monitoring/tests/monitoring) directory in the specifications repo to our `testdata/command-monitoring` directory. Looks like their copy had no new line at the end and ours did. It shouldn't make a difference to our test runner.

**Comment by benjirewis** [comment]
Merging as failures seem to be due to https://jira.mongodb.org/browse/BF-26885.

**Pull request merged**